### PR TITLE
Update fan coil and PT unit HVACTemplate inputs for Single Zone VAV control

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
Follow up to #6393 SZVAV for PT units.
Update ExpandObjects.f90, Energy+.idd.in, HVACTemplate example files for PTAC and PTHP, and IO doc.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [x] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
TO DO: 
 - [ ] Figure out why fan coil unit shows RootSolver failures when ASHRAE90VariableFan is chosen in HVACTemplate:Zone:FanCoil.
 - [x] Transition rules (not yet committed)

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 14 days since this pull request was last updated.

File: testfiles/HVACTemplate-5ZonePTAC-DOAS.idf [hunk 1]
```diff
@@ -1761,7 +1761,8 @@
     ,                        !- Design Specification Zone Air Distribution Object Name
     ,                        !- Baseboard Heating Type
     ,                        !- Baseboard Heating Availability Schedule Name
-    ;                        !- Baseboard Heating Capacity {W}
+    ,                        !- Baseboard Heating Capacity {W}
+    None;                    !- Capacity Control Method
```
Comment (rraustad) [inline testfiles/HVACTemplate-5ZonePTAC-DOAS.idf:1765]: @mjwitte should I be adding this last input to the example files? If this input didn't exist I could just default to None in ExpandObjects? Not real sure of the ExpandObjects code but it looks like all inputs need to be present. It's just a thought about num required fields.

**Comment by nrel-bot-2** [comment]
@rraustad @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad @lgentile it has been 14 days since this pull request was last updated.

File: testfiles/HVACTemplate-5ZonePTAC-DOAS.idf (hunk 1 above)
Comment (rraustad) [reply testfiles/HVACTemplate-5ZonePTAC-DOAS.idf:1765]: @mjwitte never got a response here. Do I need to add this input field or should ExpandObjects handle the missing field? It looks like HVACTemplates don't require all fields so I guess I answered my own question. Take these back out.

Comment (mjwitte) [reply testfiles/HVACTemplate-5ZonePTAC-DOAS.idf:1765]: @rraustad If I remember correctly, at one point all fields were expected to be present but that was relaxed later on.  Just be sure to test it both ways and make sure that min-fields in the IDD matches what expandobjects expects.

**Comment by Myoldmopar** [comment]
@mjwitte I see you just commented, so I will assume you will be the reviewer on this and I'll move on to others.

**Comment by rraustad** [comment]
The logic in fan coil is just munged up. Take this for example:

    if (CurDeadBandOrSetback(ZoneNum) || AirMassFlow < SmallMassFlow) UnitOn = false;

If predictor thinks the zone is in the deadband and the fan coil has flow, then OA can exceed the zone load and push the zone past the SP. In this case, the unit is on. If no OA, then no problem! I'm still looking for a way around this but it looks like I have to change all the logic around how this model runs. Not what I'd hoped for and not of the "Do no harm" mantra.

**Comment by mjwitte** [comment]
@rraustad I never have understood the need for CurDeadBankOrSetback anymore.  LoadToHeatingSetpoint and LoadToCoolingSetpoint should be all that's necessary.

**Comment by rraustad** [comment]
HVACTemplate files for FanCoil and PTUnit now work when choosing to model SZVAV. Reviewed idf's and testing annual runs.

**Comment by Myoldmopar** [comment]
@rraustad I had to clean up a bunch of warnings in this branch, some related to your work, some unrelated.  But at this point, with your branch, the build is 100% green on my system, and CI should agree.  If no problems exist as CI works on this this morning then we should be good to go in.

**Comment by Myoldmopar** [comment]
@rraustad could you take a look at the single failing unit test?  It doesn't look like an anomaly, it looks like a proper failure.  If you make a fix for that, make sure to pull in develop before pushing.  Thanks!

**Comment by Myoldmopar** [comment]
OK I fixed the failing unit test, pulled develop, and the whole suite of unit tests pass locally.  Mac may still complain about the one it's already complaining about it but that's nothing related to this, and I'm currently building on my Mac to figure that one out.  I'm pushing my changes up to this branch and merging.

**Pull request merged**